hummingbot/strategy/spot_perpetual_arbitrage/spot_perpetual_arbitrage.py

```diff
@@ -97,7 +98,9 @@ def init_params(self,
         self._strategy_state = StrategyState.Closed
         self._ready_to_start = False
         self._last_arb_op_reported_ts = 0
+
         perp_market_info.market.set_leverage(perp_market_info.trading_pair, self._perp_leverage)
+        perp_market_info.market.set_position_mode(PositionMode.ONEWAY)
```

Comment:
Whether a connector is ready doesn't seem to be checked anywhere in the application. Should I add waiting for `ready()` to the `_initialize_markets()` method in the `HummingbotApplication` ?

Reply:
Yes, in PMM there's an explicit check with the `all_markets_ready` flag:

![image](https://user-images.githubusercontent.com/7762229/163554946-5a6ec0d4-2ac0-4398-8487-2e10a4e9f93c.png)

But just like in the spot-perpetual MM, in PMM also `_initialize_markets()` is being called in the `start.py`.

So it would make more sense to put the `all_markets_ready` check to the `_initialize_markets()` method, than to recreate it in every strategy.

Reply:
I think the only use for that check is on the PMM Scripts. In fact I am not sure why we have that method implemented in other strategies. I don't think it should be something general to all strategies (at least I don't see the use for it)

Reply:
So should I add the same method also to the spot-perpetual, since originally in this thread you requested to wait until the connector is ready before setting the leverage and position mode?

Reply:
Yes, let's add that if the strategy needs to ensure a particular position mode for operating